lib/quicly.c

```diff
@@ -2781,6 +2805,10 @@ static int commit_send_packet(quicly_conn_t *conn, quicly_send_context_t *s, enu
 
     ++conn->egress.packet_number;
     ++conn->super.stats.num_packets.sent;
+    if (get_epoch(*s->target.first_byte_at) == QUICLY_EPOCH_INITIAL || get_epoch(*s->target.first_byte_at) == QUICLY_EPOCH_HANDSHAKE)
+        ++conn->super.stats.num_packets.handshake_sent;
+    if (get_epoch(*s->target.first_byte_at) == QUICLY_EPOCH_0RTT)
+        ++conn->super.stats.num_packets.zerortt_sent;
```

Comment:
As stated above, maybe it would be better to change the type of `num_packets.sent` to `uint64_t [QUICLY_NUM_EPOCHS]` and increment the per-epoch counter.